frontend/src/components/ScheduleSearch.tsx

```diff
@@ -0,0 +1,254 @@
+import { useCallback, useEffect, useRef, useState } from "react";
+import { useVirtual } from "react-virtual";
+import { useCombobox, useMultipleSelection } from "downshift";
+import { PencilAltIcon, SearchIcon } from "@heroicons/react/solid";
+import resolveConfig from "tailwindcss/resolveConfig";
+import tailwindConfig from "../../tailwind.config.js";
+import { useAppDispatch, useAppSelector } from "../app/hooks";
+import { userSlice } from "../app/user";
+import {
+  fetchAllCourses,
+  fetchCourseInfos,
+  fetchFCEInfos,
+} from "../app/courses";
+
+const fullConfig = resolveConfig(tailwindConfig);
+
+const CourseCombobox = ({ onSelectedItemsChange }) => {
+  const [inputValue, setInputValue] = useState("");
+  const listRef = useRef();
+  const dispatch = useAppDispatch();
+
+  const allCourses = useAppSelector((state) => state.courses.allCourses);
+
+  useEffect(() => {
+    dispatch(fetchAllCourses());
+  }, []);
```

Comment:
Codacy has a fix for the issue: [React Hook useEffect has a missing dependency: 'dispatch'. Either include it or remove the dependency array.](https://app.codacy.com/gh/ScottyLabs/course-tool/pullRequest?prid=9667658)
```suggestion
  }, [dispatch]);
```